Mathlib/Algebra/Order/AddTorsor.lean

```diff
@@ -44,65 +53,95 @@ open Function
 variable {G P : Type*}
 
 /-- An ordered vector addition is a bi-monotone vector addition. -/
-class OrderedVAdd (G P : Type*) [LE G] [LE P] [VAdd G P] : Prop where
+class IsOrderedVAdd (G P : Type*) [LE G] [LE P] [VAdd G P] : Prop where
   protected vadd_le_vadd_left : ∀ a b : P, a ≤ b → ∀ c : G, c +ᵥ a ≤ c +ᵥ b
   protected vadd_le_vadd_right : ∀ c d : G, c ≤ d → ∀ a : P, c +ᵥ a ≤ d +ᵥ a
 
-instance OrderedAddCommMonoid.toOrderedVAdd [OrderedAddCommMonoid G] : OrderedVAdd G G where
-  vadd_le_vadd_left _ _ := add_le_add_left
-  vadd_le_vadd_right _ _ h a := add_le_add_right h a
+@[deprecated (since := "2024-07-15")] alias OrderedVAdd := IsOrderedVAdd
 
-instance OrderedVAdd.toCovariantClassLeft [LE G] [LE P] [VAdd G P] [OrderedVAdd G P] :
-    CovariantClass G P (· +ᵥ ·) (· ≤ ·) where
-  elim := fun a _ _ bc ↦ OrderedVAdd.vadd_le_vadd_left _ _ bc a
+/-- An ordered scalar multiplication is a bi-monotone scalar multiplication. -/
+@[to_additive]
+class IsOrderedSMul (G P : Type*) [LE G] [LE P] [SMul G P] : Prop where
```

Comment:
How does this differ from the existing [`OrderedSMul`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Module/OrderedSMul.html#OrderedSMul)?

Reply:
They seem to be almost completely different:

- `OrderedSMul` is about strict inequality, while `IsOrderedSMul` is about `LE`
- `OrderedSMul` requires `G` to be an ordered semiring and `P` to have `OrderedAddCommMonoid` structure.  However, the algebraic structures do not appear in the defining `Prop`s.
- `OrderedSMul` conditions only trigger for strictly positive elements of `G`, while `IsOrderedSMul` more or less requires `G` to be made of non-negative elements.  I'm really only interested in the additive version, where this condition is meaningless.

I am a little concerned about the near collision in names, but the `OrderedSMul` module docstring suggests that it may be deprecated at some point in the future.

Reply:
Maybe @YaelDillies has thoughts

Reply:
To explain, this is the same distinction as between `OrderedCommMonoid` and `OrderedCommMonoidWithZero`.

I indeed hope `OrderedSMul` will go away soon, but it's a bit more complicated than I had originally hoped